### PR TITLE
Added EitherMatcher in place of Left/RightMatcher

### PR DESCRIPTION
**Pull request by 7h3kk1d**
Addressing https://github.com/palatable/lambda/issues/80

Do we want javadocs for the matchers or tests of the matchers themselves?

### CODE REVIEW

**Comment by jnape** [comment]
Eventually the matchers should probably be pulled out into a separate project from the tests, at which point they'll certainly need javadoc and probably their own tests. For now, just the matcher is fine.

File: src/test/java/testsupport/matchers/EitherMatcher.java [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+    @Override
+    public void describeTo(Description description) {
+        matcher.match(l -> io(() -> description.appendText("Left value of "))
+                          .flatMap(constantly(io(() -> l.describeTo(description)))),
+                      r -> io(() -> description.appendText("Right value of "))
+                          .flatMap(constantly(io(() -> r.describeTo(description)))))
+            .unsafePerformIO();
+    }
+
+    public static <L, R> EitherMatcher<L, R> isEitherThat(Either<Matcher<L>, Matcher<R>> matcher) {
```
Comment (jnape) [inline src/test/java/testsupport/matchers/EitherMatcher.java]: Should this be `private`? It seems like this is just for internal construction, as it's fully specified by `isLeftThat` and `isRightThat`. Or were you thinking of something else here?

Comment (7h3kk1d) [reply src/test/java/testsupport/matchers/EitherMatcher.java]: I didn't have a specific need in mind but it seemed like there could be a need I couldn't think of. I'm fine making it private though if it seems unnecessary.

**Comment by jnape** [comment]
Merged in b2d70bbb.

**Pull request closed without merging**